tests/test_recipient_csv.py

```diff
@@ -162,6 +162,59 @@ def test_get_rows(file_contents, template_type, expected):
             assert rows[index].get(key).data == value
 
 
+def test_get_rows_does_no_error_checking_of_rows_or_cells(mocker):
+    has_error_mock = mocker.patch.object(Row, 'has_error')
+    has_bad_recipient_mock = mocker.patch.object(Row, 'has_bad_recipient')
+    has_missing_data_mock = mocker.patch.object(Row, 'has_missing_data')
+    cell_recipient_error_mock = mocker.patch.object(Cell, 'recipient_error')
+
+    recipients = RecipientCSV(
+        """
+            email address, name
+            a@b.com,
+            a@b.com, My Name
+            a@b.com,
+
+
+        """,
+        template_type='email',
+        placeholders=['name'],
+        max_errors_shown=3
+    )
+
+    rows = recipients.get_rows()
+    for i in range(3):
+        assert next(rows).recipient == 'a@b.com'
+
+    assert has_error_mock.called is False
+    assert has_bad_recipient_mock.called is False
+    assert has_missing_data_mock.called is False
+    assert cell_recipient_error_mock.called is False
+
+
+def test_get_rows_only_iterates_over_file_once(mocker):
+    row_mock = mocker.patch('notifications_utils.recipients.Row')
+
+    recipients = RecipientCSV(
+        """
+            email address, name
+            a@b.com,
+            a@b.com, My Name
+            a@b.com,
+
+
+        """,
+        template_type='email',
+        placeholders=['name'],
+    )
+
+    rows = recipients.get_rows()
+    for i in range(3):
+        next(rows)
+
+    assert row_mock.call_count == 3
```

Comment:
Should test to make sure that when you only call `get_rows()` it doesn’t build up a list in memory:
```suggestion
    assert row_mock.call_count == 3
    assert recipients.rows_as_list == None
```